main.go

```diff
@@ -27,21 +27,36 @@ import (
 
 const (
 	extensionName = "vault-lambda-extension"
+	vaultLogLevel = "VAULT_LOG_LEVEL" // Optional, one of TRACE, DEBUG, INFO, WARN, ERROR, OFF
 )
 
 func main() {
-	logger := log.New(os.Stderr, fmt.Sprintf("[%s] ", extensionName), log.Ldate|log.Ltime|log.LUTC)
+	hclog.LevelFromString(os.Getenv(vaultLogLevel))
+	logger := hclog.New(&hclog.LoggerOptions{
+		Level: hclog.Debug,
+	})
```

Comment:
Looks like we need to use the result from `hclog.LevelFromString()` to set the level,  and default to info if its not set? Something like:

```suggestion
	level := hclog.LevelFromString(os.Getenv(vaultLogLevel))
	logger := hclog.New(&hclog.LoggerOptions{
		Level: hclog.Info,
	})
	if level != hclog.NoLevel {
		logger.SetLevel(level)
	}
```

Reply:
Doh, sorry that was very sloppy.

Reply:
Fixed - `hclog` handles switching to the default level if `NoLevel` is set internally.